### ISSUE TITLE
About the hyperparameters setting in BYOL

### ISSUE THREAD

**Issue by scnuhealthy** [open-mmlab/mmselfsup#47]
This is an outstanding project and sorry to bother you again.
I notice that the hyparameters setting reported in byol paper are different between training in 1000 epochs and training in 300 epochs (0.996 compared to 0.99 in exponential moving average parameter, 0.2 compared to 0.3 in learning rate and 1.5*10e-6 compared to 10e-6). However, this project reproduces byol in terms of 200 epochs while it uses the hyparameters setting in 1000 epochs. May this cause the gap between the reproduced result and that reported in the paper ?
Hoping for your reply, thanks!

**Comment by XiaohangZhan** [comment]
Thank you so much to find out that. We will try.

**Comment by mathfinder** [comment]
The "256 batchsize" experiment does not set "update_interval". However, I notice the experiment in the BYOL paper accumulates grad to avoid performance dropping. I wonder if it helps to set update_interval=16.
![image](https://user-images.githubusercontent.com/23050243/95012660-07620b80-066d-11eb-9ac3-7370c5754606.png)

**Comment by scnuhealthy** [comment]
I set batchsize=96*7, update_interval=6, to simulate the 4096 batch size. After training 300 epochs with the correct hyperparameter setting and apex, my result is 70% Top1 acc (although my linear evaluation setting has addition tricks). Notice that 96 (batch size in each machine) is a key parameter, if it reduces to 32 the performance is much worse. May I pull a request to share my config?

**Comment by XiaohangZhan** [comment]
> I set batchsize=96*7, update_interval=6, to simulate the 4096 batch size. After training 300 epochs with the correct hyperparameter setting and apex, my result is 70% Top1 acc (although my linear evaluation setting has addition tricks). Notice that 96 (batch size in each machine) is a key parameter, if it reduces to 32 the performance is much worse. May I pull a request to share my config? 

Sure.You are welcome to make the PR. Could you please test on ImageNet with our official testing script to share the results?

**Comment by scnuhealthy** [comment]
Ok, I will test it.

**Comment by scnuhealthy** [comment]
The result is about 67% if with the official testing script. The official hyperparameter setting when doing linear evaluation is easier to   result in overfitting and consume more time compared to total 60 epochs of mine. I guess the learning rate in official setting is too high and it may not suitable for every self-supervised method. But anyway,  we can reproduce BYOL with the gradient accumulation strategy.

**Comment by XiaohangZhan** [comment]
The evaluation setting follows MoCo. We may not able to change it. 67% is still a large gap to the official result of BYOL 300epoch (72.5%), even lower than the result of 200 epoch in our reproduction (bs4096). We are very grateful you can perform these experiments but an update of this result might misleading other users.

**Comment by XiaohangZhan** [comment]
> The "256 batchsize" experiment does not set "update_interval". However, I notice the experiment in the BYOL paper accumulates grad to avoid performance dropping. I wonder if it helps to set update_interval=16.
> ![image](https://user-images.githubusercontent.com/23050243/95012660-07620b80-066d-11eb-9ac3-7370c5754606.png)

Yes, since they accumulate the gradients, the bs256 setting is still bs4096. For a fair comparison with other methods, we prefer to keeping update_interval=1. However, if you want to reproduce their results, you can try to set it as 16. But note that their evaluation setting is different, either, so you may not be able to exactly reproduce it. We cannot reproduce their result for now.

**Comment by cxxgtxy** [comment]
@scnuhealthy @XiaohangZhan Looking forward to your replies for https://github.com/open-mmlab/OpenSelfSup/issues/61
